### DISCUSSION TITLE
(v1.21) SSL BIGNUM - Memory allocation failed

### DISCUSSION THREAD

**Discussion by bobveringa** [micropython/micropython#12791] [ESP32] [answered]
After upgrading to MicroPython version 1.21 I'm getting the following error when attempting to set up an SSL connection to my MQTT broker. 

```
Traceback (most recent call last):
  File "connection_tools".py", line 403, in connect
  File "lib/mqtt_as/mqtt_as.py", line 565, in connect
  File "lib/mqtt_as/mqtt_as.py", line 243, in _connect
OSError: (-17168, 'RSA - The private key operation failed : BIGNUM - Memory allocation failed')
```

This is with the following code

```
self._sock = ussl.wrap_socket(self._sock, {})
```

The error occurs right after startup (within 10 seconds), so this is not the cause of some longer-term memory leak.

Everything worked fine with version 1.20. And there seems to be ~100k of MicroPython memory available (using gc.mem_free()). 

There are some user c modules in this project, but they don't use up a lot of memory, and they are mostly to speed up calculations and compression (of very specific assets). 

What should I look at to resolve this error?

**Comment by jimmo** [comment, marked as answer]
@bobveringa I suspect this is the same underlying cause as https://github.com/orgs/micropython/discussions/12788 -- I will reply there.
